coauthors/lexerparser.go

```diff
@@ -0,0 +1,254 @@
+package coauthors
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+// ParseCoauthors receives string input directly from --with
+// parses it and returns a map of alias => coauthor?
+// for solo aliases coauthor will be an empty string.
+func ParseCoauthors(coauthorsArg string) (CoauthorsMap, error) {
+	l := newLexer(coauthorsArg)
+	p := newParser(l)
+
+	return p.parseCoauthors()
+}
+
+// CoauthorsMap is a mapping from --with supplied
+// alias/author name. Alias will always be
+// a non-empty string. Author may be an empty
+// string for solo-aliases
+type CoauthorsMap = map[Alias]Author
```

Comment:
Can we move the data structure to the main go file? So that the lexer/parser only contains the parsing? Not sure if go can handle this...

Reply:
@simonharrer We can't move the CoauthorsMap into `mob.go` because it will create a cyclic dependency, but we could import coauthors like

```go
import (
    ...
    
    . "mob.sh/coauthors"
)
```
and import coauthors' members directly into mob without the namespacing